### DISCUSSION TITLE
Need help fixing the large file committed

### DISCUSSION THREAD

**Discussion by 9ive** [community/community#53889] [Programming Help]
### Body

First, I did a commit that included a large file so it couldn’t be pushed. I moved the file to gitignore but it already made a commit. I tried to fix it last time and got reverse back to the old branch so all my works gone, but I already recovered it. Now, I got some work done and would like to create a new branch without involving this large file because it's just a mock data. I still couldn’t push it since I believe that that large file committed is still there. I tried to solve it but it got worse. Can anyone please help me with the problem?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by orbitgw** [comment]
Try to set the git post buffer and recommit it.
```bash
git config --global http.postBuffer 524288000
```

**Comment by orbitgw** [comment]
Try to set the git post buffer and recommit it.
```bash
git config --global http.postBuffer 524288000
```

**Comment by hrishikesh-kocharekar-dev** [comment]
Start by running git log to find the commit hash of the commit that added the large file.

Run git rebase -i HEAD~N, replacing N with the number of commits back you need to go to reach the commit that added the large file.

This should open an interactive rebase window that shows a list of commits. Find the commit that added the large file and change the word "pick" to "edit" next to it.

Save and close the file, and Git will start the rebase process.

When Git stops at the "edit" commit, run git rm --cached path/to/large_file to remove the large file from the commit, and then run git commit --amend --no-edit to update the commit.

Finally, run git rebase --continue to finish the rebase process.

Once the rebase is complete, you should be able to push your changes to the remote repository.

To address the second issue, you can create a new branch without including the large file by running git checkout -b new-branch to create a new branch, and then removing the large file from the branch using git rm path/to/large_file. Be sure to commit these changes, and then you should be able to push the new branch to the remote repository without including the large file.

**Reply by 9ive** [reply]
The problem is after I committed the large file, I already had made a process. If I run git rebase will my work disappear?

**Reply by airtower-luna** [reply]
The `-i` in `git rebase -i` is for "interactive". You can choose to edit, keep, or discard commits you made, commit by commit. If you choose "edit" the rebase will pause at the point right after the commit you want to edit, and you can make the changes you want (`git rm`, `git add`, etc.) and then *amend* the commit instead of creating a new one (`git commit --amend`). [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) describes how to do that in detail.

**Reply by SaulRdz** [reply]
That worked for me, thanks
